engine/difficulty.h

```diff
@@ -0,0 +1,35 @@
+#pragma once
+
+#include <string>
+
+namespace pong
+{
+
+class Difficulty
+{
+public:
+    enum class Level
+    {
+        Easy,
+        Normal,
+        Hard,
+        Insane
+    };
+
+    static Level GetLevel();
+    static void SetLevel(Level level);
+
+private:
+    Difficulty() = delete;
+    Difficulty(const Difficulty&) = delete;
+    Difficulty& operator=(const Difficulty&) = delete;
+    Difficulty(Difficulty&&) = delete;
+    Difficulty& operator=(Difficulty&&) = delete;
+    ~Difficulty() = delete;
+
+    static Level sLevel;
+};
+
+std::string DifficultyString(Difficulty::Level level);
```

Comment:
I would expect this to possibly be a static method of the Difficulty class, or at least in a namespace Difficulty, such that you could do

```
pong::Difficulty::to_string(level)
```